### DISCUSSION TITLE
How to add a command to the file context menu and access the selected folder as a variable?

### DISCUSSION THREAD

**Discussion by airtonix** [Taitava/obsidian-shellcommands#430] [Q&A]
### Obsidian debug information

obsidian 1.7.4

fedora 40

https://github.com/Taitava/obsidian-shellcommands/releases/tag/0.22.0

### What shell are you using?

Bash

### Please describe the problem

> How to add a command to the file context menu and access the selected folder as a variable?

I'm currently attempting to use shell command to template out a folder with a bunch of files in it.

Using Commander and Shell Commands, i was hoping i could: 

- create a shell command called `create_project`
- with a preaction that prompts for the `{{_project_name}}`
- add a script that recursively copies the folder and files to `{{ file_menu_folder }}/{{_project_name}}`
- create a commander macro: `run command: Shell Command create_project`
- Add this macro to the file context menu

so now, when i right click on a folder i get 👍🏻 

![image](https://github.com/user-attachments/assets/4bb6df8d-1216-4c20-b387-d6e27b8df5e7)

But obvious i'm missing the variable to know which folder path i clicked on in shell commands.

**Comment by Taitava** [comment]
Hi! Thank you for your question! 🙂 

The folder can be passed to the shell command, but some changes are needed:
1. AFAIK, [Commander](https://github.com/phibr0/obsidian-commander) (or any other plugin that runs command palette commands) cannot pass parameters (the folder path) to commands that it runs, as Obsidian API does not pass parameters to commands. So, the file menu item needs to be created in another way.
2. You can enable [Folder menu](https://publish.obsidian.md/shellcommands/Events/Folder+menu) event for your shell command.
3. After you have enabled the event, you can use variable [`{{event_folder_path}}`](https://publish.obsidian.md/shellcommands/Variables/%7B%7Bevent_folder_path%7D%7D) in the shell command to get the path of the right-clicked folder.

Please let me know if this helped you? 🙂
